tasks/main.yml

```diff
@@ -0,0 +1,64 @@
+---
+
+###
+### Main entrypoint
+###
+
+- name: Check if Helm binary exists
+  shell: command -v helm >/dev/null 2>&1
+  register: helm_exist
+  ignore_errors: yes
```

Comment:
`ignore_errors` is pretty nasty and should be avoided most of the times.
What about using `failed_when: helm_exist.rc == 1` to be more precise?

Reply:
I don't get the commands status code:

```bash
TASK [rolename : debug] *********************************************************************************************************************************
Friday 22 March 2019  11:54:59 +0000 (0:00:00.490)       0:00:00.594 ********** 
ok: [localhost] => {
    "_helm_exist": {
        "changed": false, 
        "cmd": "command -v helm >/dev/null 2>&1", 
        "delta": "0:00:00.254449", 
        "end": "2019-03-22 11:54:59.224023", 
        "failed": true, 
        "msg": "non-zero return code", 
        "rc": 127, 
        "start": "2019-03-22 11:54:58.969574", 
        "stderr": "", 
        "stderr_lines": [], 
        "stdout": "", 
        "stdout_lines": []
    }
}
```

And the raw:

```bash
$ command -v helm
$ echo $?
1
```

What is nasty in using `ignore_errors` for this specific case?

Reply:
`failed_when` gives you more control over why you might want to consider a task failed (in this case only if `rc == 1`). Some commands return non-zero return codes which are _ok_, and thus allow you to build logic around that. `ignore_errors` is sort of like a "catch-all". Personally, i'd prefer using a [block](https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html#id4) and treating the failures. I suppose we can go with it as is.